app/src/main/java/org/mozilla/fenix/perf/StartupFrameworkStartMeasurement.kt

```diff
@@ -0,0 +1,86 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.perf
+
+import android.os.Process
+import android.os.SystemClock
+import kotlin.math.roundToLong
+import org.mozilla.fenix.GleanMetrics.StartupTimeline as Telemetry
+
+/**
+ * A class to measure the time the Android framework executes before letting us execute our own code
+ * for the first time in Application's initializer: this value is captured in the
+ * [Telemetry.frameworkStart] probe.
+ *
+ * Since we cannot execute code at process start, this measurement does not fit within the Glean
+ * Timespan metric start/stop programming model so it lives in its own class.
+ */
+internal class StartupFrameworkStartMeasurement(
+    private val stat: Stat = Stat(),
+    private val telemetry: Telemetry = Telemetry,
+    private val getElapsedRealtimeNanos: () -> Long = SystemClock::elapsedRealtimeNanos
+) {
+
+    private var isMetricSet = false
+
+    private var applicationInitNanos = -1L
+    private var isApplicationInitCalled = false
+
+    fun onApplicationInit() {
+        // This gets called from multiple processes: don't do anything expensive. See call site for details.
+        //
+        // In the main process, there are multiple Application impl so we ensure it's only set by
+        // the first one.
+        if (!isApplicationInitCalled) {
+            isApplicationInitCalled = true
+            applicationInitNanos = getElapsedRealtimeNanos()
+        }
+    }
+
+    /**
+     * Sets the values for metrics to record in glean.
+     *
+     * We defer these metrics, rather than setting them as soon as the values are available,
+     * because they are slow to fetch and we don't want to impact startup.
+     */
+    fun setExpensiveMetric() {
+        // The application is only init once per process lifetime so we only set this value once.
+        if (isMetricSet) return
+        isMetricSet = true
+
+        if (applicationInitNanos < 0) {
+            telemetry.frameworkStartError.record()
```

Comment:
Do you really need an event for this? You only expect this to happen _once_ per application run, right? I'd go with a `boolean` metric type instead, and just flip it to true here. See [our docs](https://mozilla.github.io/glean/book/user/adding-new-metrics.html) for a discussion on why `event` is not the right fit in this case.

Reply:
I didn't realize that page existed – I just clicked directly into the different metric types when looking between them, which is why overlooked the boolean option.

I take it:

> Importantly, events are the most expensive metric type to record, transmit, store and analyze, so they should be used sparingly, and only when none of the other metric types are sufficient for answering your question.

Is why I don't want `event`. `boolean` wfm.

Reply:
@Dexterp37 I wasn't sure which you wanted:
- frameworkStartError to be false on all runs and switched to true when there's an error
- unset on all runs and set true when there's an error

I flagged you for re-review to double-check my approach.